recipe/shim/shim.c

```diff
@@ -0,0 +1,36 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#define MAX_PATHLEN 1024
+
+/*
+  This program is a simple binary shim that replaces the following shell script:
+
+  #!/bin/bash
+  exec $PREFIX/bin/python "$@"
+
+  This is needed because MacOS security protections won't let a shell script
+  be the interpreter for another shell script in the shebang.  With this
+  shim in place, scripts like cython and f2py can bounce through
+  this executable instead of bouncing through a bash script.
+*/
+
+int main(int argc, char **argv) {
+    char *prefix;
```

Comment:
Use something like `char *prefix = @PREFIX@` and replace the value in build.sh. We want the PREFIX value at build time which is `BUILD_PREFIX` at downstream package build time.

Reply:
I'm confused about this suggestion, and don't know how this would work/what substitution you're suggesting.  I was just trying to replace the current shell script https://github.com/conda-forge/cross-python-feedstock/blob/main/recipe/activate-cross-python.sh#L53-L54 with a c version that could be pre-built and pick up `$PREFIX`, and then let the activator copy it to the right place.  But at the same time it's possible that I got something mixed up and this executable is being built for the wrong platform.

Reply:
I'm suggesting we use `echo "exec $BUILD_PREFIX/venv/cross/bin/python \"\$@\"" >> $BUILD_PREFIX/bin/python` instead so that we don't rely on environment variables.

Reply:
I still don't understand.  We can't do the exec like like because macos doesn't like it.  But I think what we actually want is to use the `BUILD_PREFIX` env var (not the `PREFIX`) and then we end up with a binary shim that goes to `$BUILD_PREFIX/venv/build/bin/python` and we're in business.  And this can be pre-compiled and then just be copied in the activation script, right?  Or am I missing something else?  That is, I don't see how to get away from the env vars, and that's how this works at the moment.

Reply:
PREFIX at the time at the time of this package being built will be the `BUILD_PREFIX` when building a downstream recipe. So, we can hard code PREFIX in side this binary and then we won't need to use any environment variables. This is what was done in the rust recipe.

Reply:
Quite frankly, I didn't understand the rust code.  It seems to be relying on some magic about installation time rewriting of something or other?  Or maybe I'm wrong?

I thought that using the runtime env var, which is what the current bash script on main does, would be simpler.  (Certainly it is easier for me to understand.)

But if we need code that is pre-compiled and then somehow magically knows the install-time prefix then I guess I would suggest using the rust PR ... certainly I'm not the one to write it in c.